pkg/services/librarypanels/librarypanels_test.go

```diff
@@ -15,78 +16,251 @@ import (
 )
 
 func TestCreateLibraryPanel(t *testing.T) {
-	libraryPanelScenario(t, "When an admin tries to create a library panel", "then it should fail if library panel already exists", func(t *testing.T) {
-		lps, context := setupTestEnv(t, models.ROLE_ADMIN, map[string]string{})
-		command := getCreateCommand(1, "Text - Library Panel")
+	t.Run("When an admin tries to create a library panel that already exists", func(t *testing.T) {
+		t.Run("then it should fail", func(t *testing.T) {
+			testScenario(t, getTestContext(), func(t *testing.T, ctx *testContext) {
```

Comment:
I think that instead of this `getTextContext()` abstraction, it would be more Go style to provide a config object, with eventual parameters diverging from the defaults.

Reply:
Will look into this in my next commit as well.

Reply:
Not sure I understand this comment. Testscenario receives a pointer to a struct not a function. I invoked the function because I didn't see the point of declaring a context variable. Now I have declared an extra variable for each test instead.